### PR TITLE
ORD-4 Refactor Entities

### PR DESCRIPTION
**Pull request by kabirbaidhya**
Refactor Entities

### CODE REVIEW

**Comment by mesaugat** [comment]
Just got this: Doctrine can convert camel case to underscore by simply doing this: https://github.com/codeliner/php-ddd-cargo-sample/blob/master/config/autoload/global.php

File: module/Order/src/Order/Entity/ItemRepository.php [hunk 1]
```diff
@@ -8,7 +8,10 @@
 
 class ItemRepository extends Repository
 {
-    public function fetchList($offset = Repository::PAGINATION_OFFSET_START, $max = Repository::PAGINATION_MAX_ROWS)
+    const PAGINATION_MAX_ROWS = 5;
```
Comment (lf-kailashraj) [inline module/Order/src/Order/Entity/ItemRepository.php:11]: move this to a common config file.

File: module/Order/src/Order/Entity/Traits/ArraySerializableTrait.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Order\Entity\Traits;
+
+
+trait ArraySerializableTrait
```
Comment (lf-kailashraj) [inline module/Order/src/Order/Entity/Traits/ArraySerializableTrait.php:6]: if this trait can be shared with other entities then we should not place it inside order/\* directory.

**Comment by kabirbaidhya** [comment]
It seems that Zend emphasizes the use of module for everything. So, we should create a separate common/shared module for such code in order to use them in other module as per our need. 
We will call this a Foundation or Common module. 

Reference: http://stackoverflow.com/questions/15561309/zf2-create-common-lib-for-database-access-across-all-modules
http://framework.zend.com/manual/2.0/en/modules/zend.module-manager.intro.html

**Comment by mesaugat** [comment]
:+1:

**Pull request merged**